### PR TITLE
Empty cache for lm_head

### PR DESCRIPTION
**Pull request by hkvision**

### CODE REVIEW

**Comment by hkvision** [comment]
Baichuan2-7b can run ~3900 input with empty cache.
tiled dequant is always disabled (threshold set to max int) on arc, shall we enable it under similar conditions on arc? @jason-dai

File: python/llm/src/bigdl/llm/transformers/low_bit_linear.py [hunk 1]
```diff
@@ -584,6 +584,13 @@ def forward(self, x: torch.Tensor):
                 # current workaround to reduce first token latency of fp32 input
                 # sometimes fp16 cause nan and training instability
                 # disable the conversion when training
+
+                # empty_cache on arc for models with large vocabulary (baichuan/qwen)
+                # for long input sequences.
+                do_empty_cache = self.weight._shape[0]*self.weight._shape[1] >= 70000*4096 \
```
Comment (cyita) [inline python/llm/src/bigdl/llm/transformers/low_bit_linear.py:590]: May be make `get_xpu_device_type(x0) == "arc" and self.weight._shape[0]*self.weight._shape[1] >= 70000*4096` as an attribute of LowBitLinear?

**Comment by cyita** [comment]
We can provide a method for PVC to use this optimization.

File: python/llm/src/bigdl/llm/transformers/low_bit_linear.py (hunk 1 above)
Comment (hkvision) [reply python/llm/src/bigdl/llm/transformers/low_bit_linear.py]: Updated, take a look?

**Review by cyita — APPROVED** [review]
LGTM

**Comment by cyita** [comment]
Please resolve the ut errors.

File: python/llm/src/bigdl/llm/transformers/low_bit_linear.py [hunk 2]
```diff
@@ -520,6 +520,12 @@ def __init__(self, input_features, output_features, qtype, bias=True,
         self.mp_group = mp_group
         self.compute_dtype = None  # only for training
         self.enable_xetla = enable_xetla
+        # empty_cache (by default on arc) for models with large vocabulary (e.g. baichuan/qwen)
+        # for long input sequences.
+        self.low_memory_mode = \
+            self.weight._shape[0]*self.weight._shape[1] >= 70000*4096 and \
+            (get_xpu_device_type(self.weight.data) == "arc" or
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/low_bit_linear.py:527]: this will be false when the layer is first created

Comment (hkvision) [reply python/llm/src/bigdl/llm/transformers/low_bit_linear.py]: correct. Updated. Will detect the device at the first forward and do only once.

File: python/llm/src/bigdl/llm/transformers/low_bit_linear.py [hunk 3]
```diff
@@ -520,8 +520,17 @@ def __init__(self, input_features, output_features, qtype, bias=True,
         self.mp_group = mp_group
         self.compute_dtype = None  # only for training
         self.enable_xetla = enable_xetla
+        # empty_cache (by default on arc) for models with large vocabulary (e.g. baichuan/qwen)
+        # for long input sequences.
+        self.device = None  # detected only once in the first forward
+        self.low_memory_mode = self.in_len * self.out_len >= 70000*4096
```
Comment (hkvision) [inline python/llm/src/bigdl/llm/transformers/low_bit_linear.py:526]: may further tune this value

**Review by lalalapotter — APPROVED** [review]
LGTM.

**Review by lalalapotter — APPROVED** [review]
LGTM.

**Comment by hkvision** [comment]
No performance issues found when testing on Arc and MTL.

**Comment by hkvision** [comment]
baichuan2-13b can run 1k-128, will add it into nightly perf; can't run 2k, max now is ~1k4.

**Pull request merged**